src/net.cpp

```diff
@@ -1213,7 +1276,22 @@ bool V2Transport::ReceivedBytes(Span<const uint8_t>& msg_bytes) noexcept
 
 std::optional<std::string> V2Transport::GetMessageType(Span<const uint8_t>& contents) noexcept
 {
-    if (contents.size() < 13 || contents[0] != 0) return std::nullopt; // Short encoding not yet supported
+    if (contents.size() == 0) return std::nullopt; // Empty contents
+    uint8_t first_byte = contents[0];
+
+    if (first_byte != 0) {
+        // Short (1 byte) encoding.
+        if (first_byte < std::size(V2_MESSAGE_IDS)) {
+            // Valid short message id.
+            contents = contents.subspan(1);
+            return V2_MESSAGE_IDS[first_byte];
+        } else {
+            // Unknown short message id.
+            return std::nullopt;
+        }
+    }
+
+    if (contents.size() < 13) return std::nullopt; // Long encoding needs at least 13 bytes
```

Comment:
nit-like: For the long encoding processing, it might be worthwhile to advance for one byte in the contents span first, to work with the remaining 12 bytes and avoid the plus ones everywhere below (especially for the array indices in the while loops)?
```suggestion
    contents = contents.subspan(1);
    if (contents.size() < 12) return std::nullopt; // Long encoding needs at least 12 bytes (after the short message id)
```